apps/ejabberd/src/mod_mam.erl

```diff
@@ -666,7 +670,8 @@ is_interesting(Host, LocJID, RemJID, ArcID) ->
 %% ----------------------------------------------------------------------
 %% Backend wrappers
 
--spec archive_id_int(ejabberd:server(), ejabberd:jid()) -> integer().
+-spec archive_id_int(ejabberd:server(), ejabberd:jid()) ->
```

Comment:
I agree, good catch :) I just fixed the spec here, didn't want to touch the function name in this PR.